src/IRac.cpp

```diff
@@ -500,31 +508,37 @@ void IRac::argo(IRArgoAC *ac,
 /// @param[in] on The power setting.
 /// @param[in] mode The operation mode setting.
 /// @param[in] degrees The set temperature setting in degrees Celsius.
+/// @param[in] sensorTemp The room (iFeel) temperature sensor reading in degrees
+///                       Celsius.
 /// @param[in] fan The speed setting for the fan.
 /// @param[in] swingv The vertical swing setting.
+/// @param[in] iFeel Whether to enable iFeel (remote temp) mode on the A/C unit.
 /// @param[in] night Enable night mode (raises temp by +1*C after 1h).
 /// @param[in] econo Enable eco mode (limits power consumed).
 /// @param[in] turbo Run the device in turbo/powerful mode.
 /// @param[in] filter Enable filter mode
 /// @param[in] light Enable device display/LEDs
 void IRac::argoWrem3_ACCommand(IRArgoAC_WREM3 *ac, const bool on,
-    const stdAc::opmode_t mode, const float degrees,
-    const stdAc::fanspeed_t fan, const stdAc::swingv_t swingv,
+    const stdAc::opmode_t mode, const float degrees, const float sensorTemp,
+    const stdAc::fanspeed_t fan, const stdAc::swingv_t swingv, const bool iFeel,
     const bool night, const bool econo, const bool turbo, const bool filter,
     const bool light) {
   ac->begin();
   ac->setMessageType(argoIrMessageType_t::AC_CONTROL);
   ac->setPower(on);
   ac->setMode(ac->convertMode(mode));
   ac->setTemp(degrees);
+  if (sensorTemp != kNoTempValue) {
+    ac->setSensorTemp(static_cast<uint8_t>(sensorTemp + 0.5));
+  }
```

Comment:
Why are you (seemingly always) setting the sensor temp half a degree higher than the requested `sensorTemp` value? This makes little sense to me.
Also, it seems like a lot of duplicate code (for every time you do it) in each method.
This goes for almost all occurances of it.

Reply:
Also, if it is to make the sensor temp be higher than the desired temp so the A/C operates, that's potentially a failure.
i.e. The A/C units might not decide to operate within +/-2 deg or something. Also, you may be assuming "Cool" mode. i.e. If the unit is in "Heat" mode, you'd want the opposite in that approach.

In principle, these methods/procedures/etc should deliver the results they have been instructed to do, not change the behavior unless needed. That allows decissions, changes, and operational logic to be done at a higher (and maybe more common) level.

But this is all a guess for why you're adding half a degree ... shrug.

Reply:
> Also, it seems like a lot of duplicate code (for every time you do it) in each method.

`std::round` would (kind-of) take care of it _(fun fact... it will be more characters to write than the `+0.5`  :) but much more proper )_.

I'd argue though the `static _cast<>()` should stay there and while this is repeated in multiple places... is **not** boilerplate (I'm of the opinion a narrowing cast should **always** be very explicit, so that it signals the "_I know what I'm doing_" to future maintainers :) ).
Creating a custom "`i8round()`" function (following boost's `iround` [[ref](https://www.boost.org/doc/libs/1_72_0/boost/math/special_functions/round.hpp)]) to get rid of the `static_cast` and implement overflow checking... is also possible, but I think it would be an overkill for this (there's no overflow check today for `degrees`, so probably is a YAGNI item).